robot/robot.go

```diff
@@ -57,6 +58,9 @@ func (r *Robot) Move(steps int) error {
 	if r.EnvMap.Map == "" {
 		return fmt.Errorf("The robot can't move because no environmental map is given")
 	}
+	if !r.canMoveToPosition(steps) {
+		return fmt.Errorf("The robot can't move %v steps because their are obstacles in the way.Here's the current map: %v", steps, mapWithRobot(r))
+	}
 	newPosition := r.Position
 	switch r.Position.Orientation {
```

Comment:
See (commit 5b935be6c2ac80a3606fabc11c440e69f4153d24).
I merged the check if the robot can move with the actual move. This kind of makes the move of the switch/case to a separate method redundant, because it's only used once now. I still like it better for readability though.